File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

import Data.List

convert :: Int -> String
convert n =
        if outStr == ""
            then show n
            else outStr
        where
            factors = [(3, "Pling"), (5, "Plang"), (7, "Plong")]
            output = map (\(f, str) -> if n `rem` f==0 then str else "" ) factors
            outStr = intercalate "" output

File: go/raindrops/raindrops.go
package raindrops

import (
	"strconv"
	"strings"
)

var factors = []int{3, 5, 7}

// Convert finds raindrops
func Convert(number int) string {

	res := make([]string, 0)

	for _, v := range factors {
		if number%v == 0 {
			switch v {
			case 3:
				res = append(res, "Pling")
			case 5:
				res = append(res, "Plang")
			case 7:
				res = append(res, "Plong")
			}
		}
	}

	if len(res) == 0 {
		return strconv.Itoa(number)
	}

	return strings.Join(res, "")
}
